sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -105,12 +108,28 @@ abstract class ColumnarOutputWriter(path: String, context: TaskAttemptContext,
    */
   def write(batch: ColumnarBatch, statsTrackers: Seq[ColumnarWriteTaskStatsTracker]): Unit = {
     var needToCloseBatch = true
+    val newBatch =
+      if (outputTimestampType == ParquetOutputTimestampType.TIMESTAMP_MILLIS.toString) {
+        new ColumnarBatch(GpuColumnVector.extractColumns(batch).map {
+          cv => {
+            cv.dataType() match {
+              case DataTypes.TimestampType => new GpuColumnVector(DataTypes.TimestampType,
+                withResource(cv.getBase()) { v =>
+                  v.castTo(DType.TIMESTAMP_MILLISECONDS)
+                })
+              case _ => cv
+            }
+          }
+        })
+      } else {
+        batch
+      }
     try {
       val writeStartTimestamp = System.nanoTime
       val writeRange = new NvtxRange("File write", NvtxColor.YELLOW)
       val gpuTime = try {
         needToCloseBatch = false
-        writeBatch(batch)
+        writeBatch(newBatch)
```

Comment:
Sorry I finally spent the time to look at where the date time rebase code goes in for writes and if you actually exercised it the code would blow up with an assertion error because it is expecting to only ever see MICROS, not MILLIS, and the cast to MILLIS is happening before this code runs.

We need to fix this, and ideally update all of the existing write tests that use timestamps to also test with MILLIS if possible instead of adding in one very targeted new test.

Reply:
I have updated the "EXCEPTION" path of the code. I have also updated the parquet_test.py to use TIMESTAMP_MILLIS.